File: Andriod/Practical/USL/app/src/main/java/com/example/usl/Network/ApiCallBack.java
package com.example.usl.Network;

import com.example.usl.Utils.Cv;

import org.greenrobot.eventbus.EventBus;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ApiCallBack<T> implements  Callback<T>{
        T t;
        public ApiCallBack(T t) {
            this.t=t;
        }

        @Override
        public void onResponse(Call<T> call, Response<T> response) {
            if(response.code()==200){
                T body=response.body();
                EventBus.getDefault().post(body);
            }else{
                EventBus.getDefault().post(Cv.TIMEOUT);
            }
        }

        @Override
        public void onFailure(Call<T> call, Throwable t) {

            EventBus.getDefault().post(Cv.TIMEOUT);
        }
}

File: Andriod/Practical/usl_kotlin/app/src/main/java/com/usl/usl/network/ApiCallBack.kt
package com.usl.usl.network

import com.usl.usl.utils.Cv
import org.greenrobot.eventbus.EventBus
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ApiCallBack<T>: Cv(),Callback<T>{

    var t: T? = null
    fun ApiCallBack(t: T) {
        this.t = t
    }

    override fun onResponse(call: Call<T>?, response: Response<T>) {
        if (response.code() == 200) {
            val body: T? = response.body()
            EventBus.getDefault().post(body)
        } else {
            EventBus.getDefault().post(super.TIMEOUT)
        }
    }

    override fun onFailure(call: Call<T>?, t: Throwable?) {
        EventBus.getDefault().post(super.TIMEOUT)
    }

}